lib/fli/FliObjHdl.cpp

```diff
@@ -0,0 +1,530 @@
+        mtiInt32T enumVal;
+        std::string::iterator valIter;
+        int i = 0;
+
+        // Pad MSB for descending vector
+        for (i = 0; i < numPad && !m_ascending; i++) {
+            iter[i] = (char)m_enum_map['0']; 
+        }
+
+        for (valIter = value.begin(); (valIter != value.end()) && (i < m_num_elems); valIter++, i++) {
+            enumVal = m_enum_map[*valIter];
+            iter[i] = (char)enumVal; 
+        }
+
+        // Fill bits a the end of the value to 0's
+        for (i = len; i < m_num_elems && m_ascending; i++) {
+            iter[i] = (char)m_enum_map['0']; 
+        }
+
+        if (m_is_var) {
+            mti_SetVarValue(get_handle<mtiVariableIdT>(), (long)iter);
+        } else {
+            mti_SetSignalValue(get_handle<mtiSignalIdT>(), (long)iter);
+        }
+    }
+
+    return 0;
+}
+
+int FliIntObjHdl::initialise(std::string &name, std::string &fq_name)
+{
+    m_num_elems   = 1;
+
+    m_val_buff = (char*)malloc(33);  // Integers are always 32-bits
+    if (!m_val_buff) {
+        LOG_CRITICAL("Unable to alloc mem for value object read buffer: ABORTING");
+        return -1;
+    }
+    m_val_buff[m_num_elems] = '\0';
+ 
+    return FliValueObjHdl::initialise(name, fq_name);
+}
+
+const char* FliIntObjHdl::get_signal_value_binstr(void)
+{
+    mtiInt32T val;
+
+    if (m_is_var) {
+        val = mti_GetVarValue(get_handle<mtiVariableIdT>());
+    } else {
+        val = mti_GetSignalValue(get_handle<mtiSignalIdT>());
+    }
+
+    std::bitset<32> value((unsigned long)val);
+    std::string bin_str = value.to_string<char,std::string::traits_type, std::string::allocator_type>();
+    snprintf(m_val_buff, 33, "%s", bin_str.c_str());
+
+    return m_val_buff;
+}
+
+long FliIntObjHdl::get_signal_value_long(void)
+{
+    mtiInt32T value;
+
+    if (m_is_var) {
+        value = mti_GetVarValue(get_handle<mtiVariableIdT>());
+    } else {
+        value = mti_GetSignalValue(get_handle<mtiSignalIdT>());
+    }
+
+    return (long)value;
+}
+
+int FliIntObjHdl::set_signal_value(const long value)
+{
+    if (m_const) {
+        LOG_ERROR("Attempted to set a constant signal/variable!\n");
+        return -1;
+    }
+
+    if (m_is_var) {
+        mti_SetVarValue(get_handle<mtiVariableIdT>(), value);
+    } else {
+        mti_SetSignalValue(get_handle<mtiSignalIdT>(), value);
+    }
+
+    return 0;
+}
+
+int FliRealObjHdl::initialise(std::string &name, std::string &fq_name)
+{
+    int buff_len = 0;
+
+    m_num_elems   = 1;
+    buff_len      = (sizeof(double) + (sizeof(*m_mti_buff) - 1)) / sizeof(*m_mti_buff);
+
+    m_mti_buff    = (mtiInt32T*)malloc(sizeof(*m_mti_buff) * buff_len);
+    if (!m_mti_buff) {
+        LOG_CRITICAL("Unable to alloc mem for value object mti read buffer: ABORTING");
+        return -1;
+    }
+ 
+    return FliValueObjHdl::initialise(name, fq_name);
+}
+
+double FliRealObjHdl::get_signal_value_real(void)
+{
+    double *iter = (double *)m_mti_buff;
```

Comment:
Would it make more sense the redefine m_mti_buff as the correct type in these sub classes of FliValueObjHandle, it would mean slightly more typing but be clearer and remove casting between types.